beaconChain/helpers/stateTransitionHelpers.ts

```diff
@@ -38,22 +39,46 @@ function shuffle<T>(values: T[], seed: hash32): T[] {
 
   // The range of the RNG places an upper-bound on the size of the list that may be shuffled.
   // It is a logic error to supply an oversized list.
-  if (valuesCount < randMax) { throw new Error("Oversized list supplied to shuffle()!"); }
+  if (!(valuesCount < randMax)) { throw new Error("Oversized list supplied to shuffle!"); }
 
   // Make a copy of the values
   const output: T[] = values.slice();
-  const source = seed; // REALLY??
-  const index = 0; // REALLY??
+  let source = toBuffer(seed);
+  let index = 0;
   while (index < valuesCount - 1) {
     // Re-hash the `source` to obtain a new pattern of bytes.
-    // TODO figure out what this hash function is in python -> JS
-    // let source = hash(source)
+    source = keccak256(source).slice(0, 32);
```

Comment:
here as well :)

Reply:
Not able to set the type here as its already been declared.

Reply:
cool makes sense!